internal/v1alpha2/controllers/bundledeployment/bundledeployment.go

```diff
@@ -46,11 +49,26 @@ type bundleDeploymentReconciler struct {
 	Scheme   *runtime.Scheme
 	Recorder record.EventRecorder
 
-	controller        crcontroller.Controller
+	controller crcontroller.Controller
+
+	// unpacker knows how to unpack and store the contents locally
+	// on filesystem for the bundle types which would be handled
+	// by this reconciler.
+	unpacker source.Unpacker
+
 	dynamicWatchMutex sync.RWMutex
 	dynamicWatchGVKs  map[schema.GroupVersionKind]struct{}
 }
 
+// Options to configure bundleDeploymentReconciler
+type Option func(bd *bundleDeploymentReconciler)
+
+func WithUnpacker(u *source.Unpacker) Option {
```

Comment:
When is it valid not to have an unpacker?

Reply:
Shouldn't we just have this as a required field in the constructor instead of an option?